main/tests/server/src/com/google/refine/grel/controls/ForRangeTests.java

```diff
@@ -27,15 +27,99 @@
 
 package com.google.refine.grel.controls;
 
+import com.google.refine.RefineTest;
+import com.google.refine.expr.*;
+import com.google.refine.model.Project;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
 
 import com.google.refine.util.TestUtils;
 
-public class ForRangeTests {
+import java.util.Properties;
+
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+
+public class ForRangeTests extends RefineTest {
 
     @Test
     public void serializeForRange() {
         String json = "{\"description\":\"Iterates over the variable v starting at \\\"from\\\", incrementing by \\\"step\\\" each time while less than \\\"to\\\". At each iteration, evaluates expression e, and pushes the result onto the result array.\",\"params\":\"number from, number to, number step, variable v, expression e\",\"returns\":\"array\"}";
         TestUtils.isSerializedTo(new ForRange(), json);
     }
+
+    @Test
+    public void testForRangeWithPositiveIncrement() throws ParsingException {
+        String test[] = { "forRange(0,6,1,v,v).join(',')", "0,1,2,3,4,5" };
+        bindings = new Properties();
+        bindings.put("v", "");
+        parseEval(bindings, test);
+    }
+
+    @Test
+    public void testForRangeWithNegativeIncrement() throws ParsingException {
+        String test[] = { "forRange(6,0,-1,v,v).join(',')", "6,5,4,3,2,1" };
+        bindings = new Properties();
+        bindings.put("v", "");
+        parseEval(bindings, test);
+    }
+
+    @Test
+    public void testForRangeWithSameStartAndEndValues() throws ParsingException {
+        String test[] = { "forRange(10,10,1,v,v).join(',')", "" };
+        bindings = new Properties();
+        bindings.put("v", "");
+        parseEval(bindings, test);
+    }
+
+    @Test
+    public void testForRangeWithFloatingPointValues() throws ParsingException {
+        String test[] = { "forRange(0,0.6,0.1,v,v).join(',')", "0.0,0.1,0.2,0.3,0.4,0.5" };
+        bindings = new Properties();
+        bindings.put("v", "");
+        parseEval(bindings, test);
```

Comment:
There was a failure in this test due to floating-point precision. Do we need to resolve this issue?

Reply:
Sorry for missing this question! Yes, all tests need to pass before we can merge. The traditional way to do floating point comparisons is to make sure two values are within a small value of each other ie `abs(f1 - f2) < epsilon` rather than `f1 == f2` but since these are string comparisons, a different solution will be needed.

Reply:
Also, I hadn't noticed that the test workflows required approval to run. Now that they've been approved once, I'm hoping they'll run automatically on future commits.

Reply:
Thank you for your response. This is my first time writing a test for floating-point numbers, so I'm not very familiar with it. Could you please give me some more advice? I've researched and only found that the solution as you said is to check if the expected and actual values are equal within a certain delta tolerance.

Reply:
Since this is operating in the string domain, one option would be to use the precision specifier on `toString()` so change the test to something like 
```
forRange(0,0.6,0.1,v,v.toString('%.3f').join(',')
```
and then adjust to expected values to something like `0.000`. This will give you enough precision to catch real errors, but not so much precision that you're bothered by differences too small to care about.

BTW, I'll be offline for the next 2-3 of days, but will check back in when I return (and @wetneb ) and the other developers can help in the mean time.

Reply:
I got it! Thanks for your help.

Reply:
Perfect!